resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/rest/kubernetes/ResourceStagingServer.scala

```diff
@@ -98,8 +98,27 @@ object ResourceStagingServer {
     } else {
       new SparkConf(true)
     }
+    val apiServerUri = sparkConf.get(RESOURCE_STAGING_SERVER_API_SERVER_URL)
+    val caCertFile = sparkConf.get(RESOURCE_STAGING_SERVER_API_SERVER_CA_CERT_FILE)
+        .map(new File(_))
+    val initialAccessExpirationMs = sparkConf.get(
+        RESOURCE_STAGING_SERVER_INITIAL_ACCESS_EXPIRATION_TIMEOUT)
+    val resourceCleanupIntervalMs = sparkConf.get(RESOURCE_STAGING_SERVER_CLEANUP_INTERVAL)
     val dependenciesRootDir = Utils.createTempDir(namePrefix = "local-application-dependencies")
-    val serviceInstance = new ResourceStagingServiceImpl(dependenciesRootDir)
+    val kubernetesClientProvider = new ResourceStagingServiceKubernetesClientProviderImpl(
+        apiServerUri, caCertFile)
+    val stagedResourcesStore = new StagedResourcesStoreImpl(dependenciesRootDir)
+    val stagedResourcesCleaner = new StagedResourcesCleanerImpl(
+      kubernetesClientProvider = kubernetesClientProvider,
+      stagedResourcesStore = stagedResourcesStore,
+      cleanupExecutorService = ThreadUtils
+          .newDaemonSingleThreadScheduledExecutor("resource-expiration"),
+      clock = new SystemClock(),
+      initialAccessExpirationMs = initialAccessExpirationMs,
+      resourceCleanupIntervalMs = resourceCleanupIntervalMs)
```

Comment:
I mainly didn't want to mix up the expiration time and the interval time here, since they're both the same type and the argument list is quite long.